### PR TITLE
[LoongArch64] Simplify flags for passing struct in registers.

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
[LoongArch64] Simplify flags for passing struct in registers.

### CODE REVIEW

**Comment by LuckyXu-HF** [comment]
This is similar to PR https://github.com/dotnet/runtime/pull/100080

@tomeksowi  , great simplification!
To reuse the duplicate codes I modified a littile bit positon of `TARGET_RISCV64` in `methodtable.cpp`.
@tomeksowi , @clamp03 Could you please take a look?

File: src/coreclr/vm/methodtable.cpp [hunk 1]
```diff
@@ -3564,7 +2925,46 @@ static bool FlattenFieldTypes(TypeHandle th, StructFloatFieldInfoFlags types[2],
     }
     return true;
 }
+#endif
+
+#if defined(TARGET_LOONGARCH64)
+int MethodTable::GetLoongArch64PassStructInRegisterFlags(TypeHandle th)
```
Comment (tomeksowi) [inline src/coreclr/vm/methodtable.cpp:2931]: Any reason why this method is LA-specific? It looks the same as RISC-V

Comment (LuckyXu-HF) [reply src/coreclr/vm/methodtable.cpp:2931]: LA and RISCV are different architecture, I think it's better to separate this function for better extensibility in the future.

**Comment by LuckyXu-HF** [comment]
@shushanhf  @jkotas  Could you please review this PR? Thanks.

File: src/coreclr/vm/methodtable.cpp (hunk 1 above)
Comment (tomeksowi) [reply src/coreclr/vm/methodtable.cpp:2931]: Yes, you're right. In fact, I'm making incoming changes in #101796 but from little I know about LoongArch ABI they will be RISC-V only.

**Review by tomeksowi — APPROVED** [review]

**Review by shushanhf — APPROVED** [review]
LGTM

**Review by jkotas — APPROVED** [review]
Thanks

**Pull request merged**